libs/api/auth/src/lib/auth-user-extractor-strategies/auth-user-extractor.strategy.ts

```diff
@@ -13,10 +13,13 @@ export class ExtractUserFromMsPrincipleHeader extends AuthUserExtractorStrategy
 		if (!headerValue) {
 			return null;
 		}
-		const payloadBuffer = Buffer.from(
-			(headerValue as string).split('.')[1],
-			'base64',
-		);
+
+		const tokenParts = headerValue.split('.');
+		if (tokenParts.length < 2) {
+			return null;
+		}
+
+		const payloadBuffer = Buffer.from(tokenParts[1], 'base64');
 		const decodedToken = JSON.parse(payloadBuffer.toString());
```

Comment:
Since you already extend the token type checking, wouldn't it also make sense to wrap the `Buffer.from` and the `JSON.parse` into a `try catch` to make sure decoding and parsing are successful too?